src/web/source.rs

```diff
@@ -19,42 +21,94 @@ enum FileType {
     RustSource,
 }
 
+/// A source file
 #[derive(PartialEq, PartialOrd)]
 struct File {
     name: String,
     file_type: FileType,
 }
 
+/// A list of source files
+///
+///
+/// Rust:
+///
+/// ```ignore
+/// FileList {
+///     metadata: MetaData {
+///         name: "rcc",
+///         version: "0.0.0",
+///         description: Some("it compiles an unholy language"),
+///         target_name: None,
+///         rustdoc_status: true,
+///         default_target: "x86_64-unknown-linux-gnu",
+///     },
+///     files: vec![
+///         File {
+///             name: "main.rs",
+///             file_type: FileType::RustSource,
+///         },
+///         File {
+///             name: "lib.rs",
+///             file_type: FileType::RustSource,
+///         },
+///     ],
+/// }
+/// ```
+///
+/// Json:
+///
+/// ```json
+/// {
+///     "metadata": {
+///         "name": "rcc",
+///         "version": "0.0.0",
+///         "description": "it compiles an unholy language",
+///         "target_name": null,
+///         "rustdoc_status": true,
+///         "default_target": "x86_64-unknown-linux-gnu",
+///     },
+///     "files": [{
+///         "name": "main.rs",
+///         "file_type_rust_source": true,
+///     }, {
+///         "name": "lib.rs",
+///         "file_type_rust_source": true,
+///     }],
+/// }
+/// ```
+///
```

Comment:
Unless this is automatically updated on changes, I would rather just have the tests. Otherwise, it's too easy for it to get out of sync with the code. The tests are pretty self-documenting anyway.